enterprise/frontend/src/metabase-enterprise/metabot/state/actions.ts

```diff
@@ -0,0 +1,55 @@
+import { createAsyncThunk } from "metabase/lib/redux";
+import { metabotAgent } from "metabase-enterprise/api";
+import type { MetabotAgentMessage } from "metabase-types/api";
+
+import { metabot } from "./reducer";
+import { getContext, getHistory } from "./selectors";
+import type { MetabotStoreState } from "./types";
+
+export const { addMessage } = metabot.actions;
+
+export const processMetabotMessages = createAsyncThunk(
+  "metabase-enterprise/metabot/processResponseMessages",
+  async (messages: MetabotAgentMessage[], { dispatch }) => {
+    for (const message of messages) {
+      if (message.type === "metabot.reaction/message") {
+        await dispatch(
+          addMessage({
+            source: "llm",
+            llm_response_type: "message",
+            message: message.message,
+          }),
+        );
+      } else {
+        console.error(
+          "Encounted unexpected message type from Metabot",
+          message,
+        );
+      }
+    }
+  },
+);
+
+export const sendMessage = createAsyncThunk(
```

Comment:
Why not use the mutation hook directly?

Reply:
The main goal is to provide a simpler interface in working with metabot. Calling `sendMessage` will decorate your request with history data, add your message to the chat history, and kick off processing the response.

I also think we'll at some point want to make sure that only one request is in-flight at a time (maybe multiple areas of the UI can interact with the agent - not just the chat bar). I think hiding RTKQuery as an implementation detail makes more sense here, but I can be swayed...

Reply:
add a custom wrapper hook?

Reply:
That's fair. I can do that.